java/compaction-status-store/src/main/java/sleeper/compaction/status/job/DynamoDBCompactionJobStatusFormat.java

```diff
@@ -74,27 +75,28 @@ public static Map<String, AttributeValue> createJobCreatedRecord(CompactionJob j
                 }).build();
     }
 
-    public static Map<String, AttributeValue> createJobStartedRecord(CompactionJob job, Instant startTime) {
-        return createJobRecord(job, UPDATE_TYPE_STARTED)
+    public static Map<String, AttributeValue> createJobStartedRecord(CompactionJob job, Instant startTime, Long timeToLive) {
+        return createJobRecord(job, UPDATE_TYPE_STARTED, timeToLive)
                 .number(START_TIME, startTime.toEpochMilli())
                 .build();
     }
 
-    public static Map<String, AttributeValue> createJobFinishedRecord(CompactionJob job, CompactionJobSummary summary) {
-        return createJobRecord(job, UPDATE_TYPE_FINISHED)
+    public static Map<String, AttributeValue> createJobFinishedRecord(CompactionJob job, CompactionJobSummary summary, Long timeToLive) {
+        return createJobRecord(job, UPDATE_TYPE_FINISHED, timeToLive)
                 .number(START_TIME, summary.getStartTime().toEpochMilli())
                 .number(FINISH_TIME, summary.getFinishTime().toEpochMilli())
                 .number(RECORDS_READ, summary.getLinesRead())
                 .number(RECORDS_WRITTEN, summary.getLinesWritten())
                 .build();
     }
 
-    private static DynamoDBRecordBuilder createJobRecord(CompactionJob job, String updateType) {
+    private static DynamoDBRecordBuilder createJobRecord(CompactionJob job, String updateType, Long timeToLive) {
         return new DynamoDBRecordBuilder()
                 .string(JOB_ID, job.getId())
                 .string(TABLE_NAME, job.getTableName())
                 .number(UPDATE_TIME, Instant.now().toEpochMilli())
-                .string(UPDATE_TYPE, updateType);
+                .string(UPDATE_TYPE, updateType)
+                .number(EXPIRY_DATE, Instant.now().toEpochMilli() + timeToLive);
```

Comment:
Can the two calls to Instant.now().toEpochMillis() be changed so that there is only 1 call? That will guarantee that the expiry date is exactly timeToLive after the update time.